src/api/flaskr/service/learn/context_v2.py

```
@@ -2058,6 +2375,13 @@ def run_inner(self, app: Flask) -> Generator[RunMarkdownFlowDTO, None, None]:
                 self._current_attend.block_position += 1
                 db.session.flush()
 
+                # Finalize TTS - synthesize remaining buffer and upload to OSS
+                if tts_processor:
+                    try:
+                        yield from tts_processor.finalize()
+                    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)